### PR TITLE
Don't allow Domain name matching production domain to be created

### PR DESCRIPTION
**Pull request by saadmk11**
closes https://github.com/readthedocs/readthedocs.org/issues/7216

### CODE REVIEW

File: readthedocs/projects/forms.py [hunk 1]
```diff
@@ -634,6 +634,19 @@ def clean_domain(self):
             domain_string = parsed.netloc
         else:
             domain_string = parsed.path
+
+        if (
+            settings.PRODUCTION_DOMAIN and
+            settings.PRODUCTION_DOMAIN in domain_string
```
Comment (ericholscher) [inline readthedocs/projects/forms.py:640]: I think we probably want a `endswith` in both of these, to reduce false positives.

File: readthedocs/projects/forms.py [hunk 2]
```diff
@@ -634,6 +634,19 @@ def clean_domain(self):
...
+        ):
+            raise forms.ValidationError('Enter a valid domain name.')
```
Comment (ericholscher) [inline readthedocs/projects/forms.py:642]: We should mention exactly why this is invalid (`{PRODUCTION_DOMAIN} is not a valid domain`

File: readthedocs/projects/forms.py [hunk 3]
```diff
@@ -634,6 +634,23 @@ def clean_domain(self):
...
+                f'{settings.PRODUCTION_DOMAIN} is not a valid domain.'
+            )
+
+        if (
+            settings.PUBLIC_DOMAIN and
+            domain_string.endswith(settings.PUBLIC_DOMAIN)
+        ):
+            raise forms.ValidationError(
+                f'{settings.PUBLIC_DOMAIN} is not a valid domain.'
+            )
```
Comment (humitos) [inline readthedocs/projects/forms.py:652]: This can be done in a for and de-duplicate this code, right?

Comment (saadmk11) [reply readthedocs/projects/forms.py]: Okay :+1:

**Comment by humitos** [comment]
Thanks @saadmk11, you rock!

**Pull request merged**